### PR TITLE
feat: CLI Cobra Support

### LINKED ISSUE #9: CLI Cobra Support
you will also need an administration tool to manage, issue, and verify verifiable credentials.

### PR DESCRIPTION
**Pull request by DRK3**
Closes #9 

Created basic starter files, which provides a foundation for future commits to build off of. Allows one to start an edge-node instance via command line or environment variables. Right now it simply creates an underlying agent that runs on the user-defined host url with default parameters.

Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (fqutishat) [inline cmd/main.go]: you need to move it to sub folder

Comment (DRK3) [reply cmd/main.go]: Can you elaborate? I don't understand why. What sort of sub folder?

Comment (fqutishat) [reply cmd/main.go]: https://github.com/hyperledger/aries-framework-go/tree/master/cmd

File: cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/trustbloc/edge-store/cmd/startcmd"
+)
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use: "edge-store",
```
Comment (fqutishat) [inline cmd/main.go]: edge-store is too generic name

File: cmd/main.go (hunk 1 above)
Comment (fqutishat) [reply cmd/main.go]: we will multiple cmd folder in this project

Comment (fqutishat) [reply cmd/main.go]: Example issuer-wasm cmd #18

File: go.mod [hunk 3]
```diff
@@ -5,3 +5,17 @@
 module github.com/trustbloc/edge-store
 
 go 1.13
+
+require (
+	github.com/gorilla/mux v1.7.3
```
Comment (fqutishat) [inline go.mod]: Cmd should have it their own go.mod

File: cmd/main.go (hunk 2 above)
Comment (DRK3) [reply cmd/main.go]: Would "edge-store-rest" be sufficient? My thought was that whether the instance is a verifier/issuer/user would be determined by a command line argument. So I'm not really sure what to name this one. Or I could perhaps have separate main files for issuer, verifier, etc... what do you think works best?

File: cmd/main.go (hunk 1 above)
Comment (DRK3) [reply cmd/main.go]: Gotcha. I've moved it to an edge-store-rest folder.

File: cmd/main.go (hunk 2 above)
Comment (fqutishat) [reply cmd/main.go]: I think edge-store-rest is ok for now

Comment (DRK3) [reply cmd/main.go]: renamed it to "edge-store-rest" for now, knowing that we'll likely change it later.

**Comment by DRK3** [comment]
Note to self: get codecov up and running before merging this.

File: go.mod (hunk 3 above)
Comment (talwinder50) [reply go.mod]: go.mod is still in the pkg path not separated

Comment (DRK3) [reply go.mod]: Done

File: cmd/edge-store-rest/startcmd/start.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	if isSet {
+		return value, nil
+	}
+
+	return "", errors.New("Neither " + flagName + " (command line flag) nor " + envKey +
+		" (environment variable) have been set.")
+}
+
+func startAgent(parameters *agentParameters) error {
+	if parameters.hostURL == "" {
```
Comment (fqutishat) [inline cmd/edge-store-rest/startcmd/start.go]: remove this code not needed

Comment (DRK3) [reply cmd/edge-store-rest/startcmd/start.go]: Done

File: cmd/edge-store-rest/main.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"log"
+
+	"github.com/trustbloc/edge-store/cmd/edge-store-rest/startcmd"
```
Comment (fqutishat) [inline cmd/edge-store-rest/main.go]: wrong location

Comment (DRK3) [reply cmd/edge-store-rest/main.go]: What's in the wrong location?

Comment (DRK3) [reply cmd/edge-store-rest/main.go]: Moved the order of the imports around

**Pull request merged**